commit ae780bd63b4b135c20443d208217ad4955caebe3 (refs/changes/16/2416/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-08-16T22:38:10+00:00 (2 years, 2 months ago)
    
    joyent/node-ip6addr#2 Test output when there are multiple groups of zeros
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/test/output.test.js b/test/output.test.js
index fe34fa1..62ab984 100644
--- a/test/output.test.js
+++ b/test/output.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016, Joyent, Inc.
+ * Copyright 2017, Joyent, Inc.
  */
 
 var test = require('tape').test;
@@ -59,6 +59,16 @@ test('output ipv6', function (t) {
   t.end();
 });
 
+test('output ipv6 (multiple groups of zero)', function (t) {
+  var addr = lib.parse('fd00:3:0:0:1:0:0:0');
+  t.equal(addr.toString(), 'fd00:3:0:0:1::');
+  addr = lib.parse('fd00:3:0:0:1:1:0:0');
+  t.equal(addr.toString(), 'fd00:3::1:1:0:0');
+  addr = lib.parse('fd00:3:0:0:1:1:1:0');
+  t.equal(addr.toString(), 'fd00:3::1:1:1:0');
+  t.end();
+});
+
 test('output ipv4', function (t) {
   var addr = lib.parse('1.2.3.4');
   t.equal(addr.toString(), '1.2.3.4');
